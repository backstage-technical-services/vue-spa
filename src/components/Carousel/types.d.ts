export interface CarouselImage {
  url: string;
  text?: string;
}

export interface CarouselProps {
  images: CarouselImage[];
  interval: number;
}
