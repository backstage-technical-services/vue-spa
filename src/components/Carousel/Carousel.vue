<template>
  <div class="carousel">
    <b-carousel :interval="interval * 1000" fade indicators>
      <b-carousel-slide
        :caption="image.text"
        :img-src="image.url"
        :key="image.url"
        caption-tag="p"
        v-for="image in images"
      />
    </b-carousel>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'
import { CarouselProps } from './types'

@Component
export default class Carousel extends Vue {
  @Prop() readonly images!: CarouselProps['images'];
  @Prop({ default: 10 }) readonly interval!: CarouselProps['interval'];
}
</script>

<style lang="scss" scoped>
  @import "../../assets/css/vars";

  ::v-deep .carousel-inner {
    height: 180px;
    overflow: hidden;
    width: 100%;

    ::v-deep .carousel-item {
      display: flex;
      flex-direction: column;
      align-items: center;

      ::v-deep img {
        height: auto;
        width: auto;
      }
    }

    @media (min-width: map-get($grid-breakpoints, sm)) {
      height: 255px;
    }

    @media (min-width: map-get($grid-breakpoints, md)) {
      height: 345px;
    }

    @media (min-width: map-get($grid-breakpoints, lg)) {
      height: 465px;
    }

    @media (min-width: map-get($grid-breakpoints, xl)) {
      height: 480px;
    }

    @media (min-width: map-get($grid-breakpoints, xxl)) {
      height: 570px;
    }
  }

  ::v-deep .carousel-caption {
    background: rgba(0, 0, 0, 0.5);
    bottom: 0;
    left: 0;
    padding-bottom: 60px;
    width: 100%;
    color: #aaa;

    p {
      margin-bottom: 0;
    }
  }
</style>
