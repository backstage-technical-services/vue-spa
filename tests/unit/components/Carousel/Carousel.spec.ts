import { createLocalVue, mount } from '@vue/test-utils'
import Carousel from '@/components/Carousel/Carousel.vue'
import BootstrapVue from 'bootstrap-vue'

const localVue = createLocalVue()
localVue.use(BootstrapVue)

describe('Carousel component', () => {
  it('renders without crashing', () => {
    const wrapper = mount(Carousel, {
      localVue,
      propsData: {
        images: [
          { url: '#1', text: 'Caption #1' },
          { url: '#2' }
        ]
      }
    })

    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper.element).toMatchSnapshot()
  })
})
