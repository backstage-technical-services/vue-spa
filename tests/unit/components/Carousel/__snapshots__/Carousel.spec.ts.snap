// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel component renders without crashing 1`] = `
<div
  class="carousel"
>
  <div
    aria-busy="false"
    class="carousel slide carousel-fade"
    role="region"
  >
    <div
      class="carousel-inner"
      role="list"
    >
      <div
        aria-current="true"
        aria-posinset="1"
        aria-setsize="2"
        class="carousel-item active"
        role="listitem"
      >
        <img
          class="img-fluid w-100 d-block"
          src="#1"
        />
        <div
          class="carousel-caption"
        >
          <p>
            Caption #1
          </p>
          <!---->
          <!---->
        </div>
      </div>
      <div
        aria-current="false"
        aria-posinset="2"
        aria-setsize="2"
        class="carousel-item"
        role="listitem"
      >
        <img
          class="img-fluid w-100 d-block"
          src="#2"
        />
        <!---->
      </div>
    </div>
    <!---->
    <ol
      aria-hidden="false"
      aria-label="Select a slide to display"
      class="carousel-indicators"
    />
  </div>
</div>
`;
